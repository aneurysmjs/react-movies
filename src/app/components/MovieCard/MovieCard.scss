.movieCard {
  margin-bottom: 1.5em;
  width: 20rem;
}