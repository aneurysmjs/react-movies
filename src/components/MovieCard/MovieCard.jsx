import React from 'react';
import { NavLink } from 'react-router-dom';

export default class MovieCard extends React.Component {

  constructor(props) {
    super(props);
  }

  render() {

    // Destructure the individual props
    let { movieTitle, movieGenre, country, description, id } = this.props;

    return (
      <article className="movieCard">
        <div className="card">
          <div className="card-body">
            <h4 className="card-title">{movieTitle}</h4>
            <p className="card-text text-clamp">
              {description}
            </p>
          </div>
          <ul className="list-group list-group-flush">
            <li className="list-group-item">{country}</li>
            <li className="list-group-item">{movieGenre}</li>
          </ul>
        </div>

        <NavLink to={`details/${id}`}>
          See Details
        </NavLink>
      </article>

    );

  }
  
}