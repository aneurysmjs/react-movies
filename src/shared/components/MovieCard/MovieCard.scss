.movieCard {
  margin-bottom: 1.5em;
  width: 20rem;
}

.movieCard__title {
  height: 60px;
}

.movieCard__content {
  max-height: 4rem;
  overflow: hidden;
}

.movieCard__text-clamp {
  -webkit-line-clamp: 3;
  overflow : hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-box-orient: vertical;
}