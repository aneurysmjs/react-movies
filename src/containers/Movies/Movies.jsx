import React, { Component } from 'react';
import { arrayOf, shape, string, number } from 'prop-types';

import Header from '../../components/Header/Header';
import MovieCard from '../../components/MovieCard/MovieCard';
import Searcher from '../../components/Searcher/Searcher';

export default class Movies extends Component {

  constructor() {
    super();

    /**
     * Movies's state, search term to filter the movies that match that criteria.
     */
    this.state = {
      searchTerm: ''
    };

    this.searchTermHandler = this.searchTermHandler.bind(this);
  }

  render() {

    const { movies } = this.props;

    return (
      <section>
        <Header />
        <Searcher searchTerm={this.state.searchTerm} onSearch={this.searchTermHandler} />
        <div className="d-flex align-items-start justify-content-between flex-wrap">
          {movies.filter(movie => {
            return `${movie.movieTitle} ${movie.description}`.toUpperCase().indexOf(this.state.searchTerm.toUpperCase()) >= 0;
          }).map(movie => (
            /*
              we can also `spread` the object, is like taking everything inside of `movie` and spread it out
              so is the exact same as movie-title={movie.movieTitle}, country={movie.country} for every single thing
              in `movie`, this is not part of ES feature but that's JSX mimicking what ES does and spread it out
              over the entire object.
            */
            <MovieCard {...movie} key={movie.id} />
          ))}
        </div>
      </section>
    );

  }

  /**
   * Sets state's searchTerm and filter the movies.
   *
   * @param {string} searchTerm
   * @return {void}
   */
  searchTermHandler(searchTerm) {
    this.setState({searchTerm});
  }
  
}

Movies.propTypes = {
  movies: arrayOf(shape({
    id: number,
    movieTitle: string,
    country: string,
    description: string,
    movieGenre: string
  }))
};