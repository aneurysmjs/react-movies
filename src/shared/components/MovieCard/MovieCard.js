import React, { PureComponent } from 'react';
import { NavLink } from 'react-router-dom';

import './MovieCard.scss';

export default class MovieCard extends PureComponent {

  render() {

    const { title, movieGenre, release_date, overview, id } = this.props;

    return (
      <article className="movieCard">
        <div className="card">
          <div className="card-body">
            <h4
              className="card-title movieCard__title movieCard__text-clamp"
            >
              {title}
            </h4>
            <p className="card-text text-clamp">
              {overview}
            </p>
          </div>
          <ul className="list-group list-group-flush">
            <li className="list-group-item">
              {release_date}
            </li>
            <li className="list-group-item">
              {movieGenre}
            </li>
          </ul>
          <NavLink to={`details/${id}`}>
            See Details
          </NavLink>
        </div>
      </article>

    );

  }
  
}