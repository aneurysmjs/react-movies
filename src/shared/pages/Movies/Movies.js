import React, { Component } from 'react';
import { connect } from 'react-redux';
import { arrayOf, shape, string, number } from 'prop-types';

import { setSearchTerm, getMovies } from '../../actions';
import Nav from 'components/Nav/Nav';
import MovieCard from 'components/MovieCard/MovieCard';

import './Movies.scss';

class Movies extends Component {

  /**
   * Sets state's searchTerm and filter the movies.
   *
   * @param {string} searchTerm
   * @return {void}
   */
  searchTermHandler = (searchTerm) => {
    this.props.setSearchTerm(searchTerm);
  };

  componentDidMount() {

    console.log('this.props', this.props);
    if (this.props.movies.length === 0) {
      
    }

  }

  componentDidUpdate() {

  }

  render() {

    const { searchTerm, movies } = this.props;

    return (
      <section>
        <Nav
          showSearch={true}
          searchTerm={searchTerm}
          onSearch={this.searchTermHandler}
        />
        <div className="px-3">
          <div className="d-flex align-items-start justify-content-between flex-wrap">
            {movies.filter(movie => (
              `${movie.movieTitle} ${movie.description}`.toUpperCase().indexOf(searchTerm.toUpperCase()) >= 0
            )).map(movie => (
              <MovieCard key={movie.id} {...movie} />
            ))}
          </div>
        </div>
      </section>
    );

  }

  
}

const mapStateToProps = (state) => ({
  searchTerm: state.searchTerm,
  movies: state.movies
});

const mapDispatchToProps = { 
  getMovies,
  setSearchTerm,  
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Movies);