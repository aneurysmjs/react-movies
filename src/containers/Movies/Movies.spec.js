import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import moviesData from '../../assets/json/movies';
import MovieCard from '../../components/MovieCard/MovieCard';
import Movies from './Movies';

test('Movies snapshot test', () => {

  const component = shallow(<Movies />);
  const tree = shallowToJson(component);

  expect(tree).toMatchSnapshot();

});

test('Render MovieCard for each Movie', () => {

  const component = shallow(<Movies />);
  expect(moviesData.length).toEqual(component.find(MovieCard).length);

});