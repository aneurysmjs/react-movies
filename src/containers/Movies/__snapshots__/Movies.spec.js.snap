// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies snapshot test 1`] = `
<section>
  <Searcher
    onSearch={[Function]}
    searchTerm=""
  />
  <div
    className="d-flex align-items-start justify-content-between flex-wrap"
  >
    <MovieCard
      country="Canada"
      description="sem mauris laoreet ut rhoncus aliquet pulvinar sed nisl nunc rhoncus dui"
      id={1}
      key="1"
      movieGenre="Documentary"
      movieTitle="Big Star: Nothing Can Hurt Me"
    />
    <MovieCard
      country="Cuba"
      description="ornare consequat lectus in est risus auctor sed tristique in tempus sit amet"
      id={2}
      key="2"
      movieGenre="Comedy|Musical|Romance"
      movieTitle="Flying Down to Rio"
    />
    <MovieCard
      country="Peru"
      description="non mi integer ac neque duis bibendum morbi non quam nec dui luctus rutrum nulla tellus"
      id={3}
      key="3"
      movieGenre="Comedy|Drama"
      movieTitle="Game 6"
    />
    <MovieCard
      country="China"
      description="sapien iaculis congue vivamus metus arcu adipiscing molestie hendrerit at vulputate vitae nisl aenean lectus pellentesque eget nunc donec quis"
      id={4}
      key="4"
      movieGenre="Drama"
      movieTitle="Profound Desires of the Gods (Kamigami no fukaki yokubo) "
    />
    <MovieCard
      country="Russia"
      description="viverra pede ac diam cras pellentesque volutpat dui maecenas tristique est et"
      id={5}
      key="5"
      movieGenre="Adventure|Children|Drama"
      movieTitle="Free Willy 3: The Rescue"
    />
    <MovieCard
      country="Kazakhstan"
      description="laoreet ut rhoncus aliquet pulvinar sed nisl nunc rhoncus dui vel sem sed sagittis nam congue risus semper porta volutpat"
      id={6}
      key="6"
      movieGenre="Drama"
      movieTitle="Seventh Seal, The (Sjunde inseglet, Det)"
    />
    <MovieCard
      country="Mali"
      description="elementum eu interdum eu tincidunt in leo maecenas pulvinar lobortis est phasellus sit"
      id={7}
      key="7"
      movieGenre="Fantasy|Horror|Mystery"
      movieTitle="Curse of the Demon (Night of the Demon)"
    />
    <MovieCard
      country="Portugal"
      description="integer aliquet massa id lobortis convallis tortor risus dapibus augue vel accumsan tellus nisi"
      id={8}
      key="8"
      movieGenre="Drama"
      movieTitle="Summer of Sam"
    />
    <MovieCard
      country="China"
      description="velit eu est congue elementum in hac habitasse platea dictumst morbi vestibulum"
      id={9}
      key="9"
      movieGenre="Drama"
      movieTitle="Tess of the Storm Country"
    />
    <MovieCard
      country="South Korea"
      description="velit id pretium iaculis diam erat fermentum justo nec condimentum neque sapien"
      id={10}
      key="10"
      movieGenre="Comedy|Drama|Romance"
      movieTitle="Joe Somebody"
    />
    <MovieCard
      country="Ivory Coast"
      description="rhoncus aliquet pulvinar sed nisl nunc rhoncus dui vel sem sed sagittis nam congue"
      id={11}
      key="11"
      movieGenre="Action|Animation|Horror"
      movieTitle="Hellsing Ultimate OVA Series"
    />
    <MovieCard
      country="Sweden"
      description="et magnis dis parturient montes nascetur ridiculus mus etiam vel augue vestibulum"
      id={12}
      key="12"
      movieGenre="Fantasy|Horror|Romance|Thriller"
      movieTitle="Dracula (Bram Stoker's Dracula)"
    />
    <MovieCard
      country="Thailand"
      description="nisi eu orci mauris lacinia sapien quis libero nullam sit amet turpis"
      id={13}
      key="13"
      movieGenre="Comedy|Drama|Romance"
      movieTitle="Chapter Two"
    />
    <MovieCard
      country="Portugal"
      description="eget tempus vel pede morbi porttitor lorem id ligula suspendisse"
      id={14}
      key="14"
      movieGenre="Comedy"
      movieTitle="Straight Talk"
    />
    <MovieCard
      country="Canada"
      description="cras mi pede malesuada in imperdiet et commodo vulputate justo in blandit ultrices enim lorem"
      id={15}
      key="15"
      movieGenre="Documentary"
      movieTitle="Human Failure (Menschliches Versagen)"
    />
    <MovieCard
      country="Venezuela"
      description="sit amet turpis elementum ligula vehicula consequat morbi a ipsum integer a"
      id={16}
      key="16"
      movieGenre="Drama|Fantasy|Mystery|Romance|Thriller"
      movieTitle="Dragonfly"
    />
    <MovieCard
      country="Japan"
      description="magna ac consequat metus sapien ut nunc vestibulum ante ipsum primis in faucibus orci luctus"
      id={17}
      key="17"
      movieGenre="Horror"
      movieTitle="Killer Crocodile"
    />
    <MovieCard
      country="Philippines"
      description="suscipit a feugiat et eros vestibulum ac est lacinia nisi venenatis tristique"
      id={18}
      key="18"
      movieGenre="Comedy"
      movieTitle="How to Be a Man"
    />
    <MovieCard
      country="Colombia"
      description="interdum venenatis turpis enim blandit mi in porttitor pede justo eu massa donec dapibus"
      id={19}
      key="19"
      movieGenre="Comedy|Romance"
      movieTitle="Sunday in New York"
    />
    <MovieCard
      country="Mozambique"
      description="vivamus in felis eu sapien cursus vestibulum proin eu mi nulla ac enim in tempor turpis nec"
      id={20}
      key="20"
      movieGenre="Comedy|Drama|Romance"
      movieTitle="Dish & the Spoon, The"
    />
    <MovieCard
      country="Armenia"
      description="at ipsum ac tellus semper interdum mauris ullamcorper purus sit amet"
      id={21}
      key="21"
      movieGenre="Comedy|Drama|Romance|Sci-Fi"
      movieTitle="I'm a Cyborg, But That's OK (Saibogujiman kwenchana)"
    />
    <MovieCard
      country="Indonesia"
      description="faucibus orci luctus et ultrices posuere cubilia curae duis faucibus accumsan"
      id={22}
      key="22"
      movieGenre="Documentary"
      movieTitle="The Lumière Brothers' First Films"
    />
    <MovieCard
      country="France"
      description="donec pharetra magna vestibulum aliquet ultrices erat tortor sollicitudin mi sit amet lobortis sapien"
      id={23}
      key="23"
      movieGenre="Western"
      movieTitle="Butch and Sundance: The Early Days"
    />
    <MovieCard
      country="Poland"
      description="donec odio justo sollicitudin ut suscipit a feugiat et eros"
      id={24}
      key="24"
      movieGenre="Drama|Mystery|Thriller"
      movieTitle="Fallen Idol, The"
    />
    <MovieCard
      country="China"
      description="morbi sem mauris laoreet ut rhoncus aliquet pulvinar sed nisl nunc"
      id={25}
      key="25"
      movieGenre="Crime|Drama|Film-Noir"
      movieTitle="Where the Sidewalk Ends"
    />
    <MovieCard
      country="China"
      description="sapien dignissim vestibulum vestibulum ante ipsum primis in faucibus orci luctus"
      id={26}
      key="26"
      movieGenre="Comedy|Drama"
      movieTitle="Dinner, The (Cena, La)"
    />
    <MovieCard
      country="Indonesia"
      description="a odio in hac habitasse platea dictumst maecenas ut massa quis augue luctus tincidunt nulla mollis molestie lorem quisque"
      id={27}
      key="27"
      movieGenre="Adventure|Fantasy|Sci-Fi"
      movieTitle="Tin Man"
    />
    <MovieCard
      country="Kosovo"
      description="venenatis tristique fusce congue diam id ornare imperdiet sapien urna"
      id={28}
      key="28"
      movieGenre="Crime|Drama|Thriller"
      movieTitle="Runner Runner"
    />
    <MovieCard
      country="China"
      description="nibh in lectus pellentesque at nulla suspendisse potenti cras in purus eu magna vulputate luctus"
      id={29}
      key="29"
      movieGenre="Drama"
      movieTitle="Simple Life of Noah Dearborn, The"
    />
    <MovieCard
      country="Philippines"
      description="morbi quis tortor id nulla ultrices aliquet maecenas leo odio condimentum id"
      id={30}
      key="30"
      movieGenre="Horror"
      movieTitle="976-EVIL"
    />
    <MovieCard
      country="China"
      description="nibh in hac habitasse platea dictumst aliquam augue quam sollicitudin vitae"
      id={31}
      key="31"
      movieGenre="Comedy|Drama"
      movieTitle="Self-Made Hero, A (Un héros très discret)"
    />
    <MovieCard
      country="Brazil"
      description="maecenas leo odio condimentum id luctus nec molestie sed justo pellentesque viverra"
      id={32}
      key="32"
      movieGenre="Documentary|Musical"
      movieTitle="Hecho en México"
    />
    <MovieCard
      country="Uruguay"
      description="vulputate elementum nullam varius nulla facilisi cras non velit nec nisi vulputate nonummy"
      id={33}
      key="33"
      movieGenre="Comedy"
      movieTitle="Grown Ups"
    />
    <MovieCard
      country="Russia"
      description="est quam pharetra magna ac consequat metus sapien ut nunc vestibulum ante ipsum primis"
      id={34}
      key="34"
      movieGenre="Crime|Drama|Thriller"
      movieTitle="Domino"
    />
    <MovieCard
      country="France"
      description="felis sed interdum venenatis turpis enim blandit mi in porttitor pede justo eu massa donec dapibus duis at"
      id={35}
      key="35"
      movieGenre="Drama"
      movieTitle="Friday Night (Vendredi Soir)"
    />
    <MovieCard
      country="China"
      description="primis in faucibus orci luctus et ultrices posuere cubilia curae nulla dapibus dolor vel est donec"
      id={36}
      key="36"
      movieGenre="Comedy|Horror|Sci-Fi"
      movieTitle="Alien Space Avenger"
    />
    <MovieCard
      country="Brazil"
      description="mi sit amet lobortis sapien sapien non mi integer ac neque"
      id={37}
      key="37"
      movieGenre="Drama|Fantasy|Horror|Mystery|Thriller"
      movieTitle="Gemini (Sôseiji)"
    />
    <MovieCard
      country="China"
      description="ornare consequat lectus in est risus auctor sed tristique in tempus"
      id={38}
      key="38"
      movieGenre="Action|Drama|Romance|War"
      movieTitle="Passchendaele"
    />
    <MovieCard
      country="Tajikistan"
      description="sit amet turpis elementum ligula vehicula consequat morbi a ipsum integer a nibh in quis justo maecenas rhoncus aliquam"
      id={39}
      key="39"
      movieGenre="Drama"
      movieTitle="Beijing Bicycle (Shiqi sui de dan che)"
    />
    <MovieCard
      country="Poland"
      description="primis in faucibus orci luctus et ultrices posuere cubilia curae mauris viverra diam vitae quam suspendisse potenti"
      id={40}
      key="40"
      movieGenre="Comedy|Drama"
      movieTitle="Early Years: Erik Nietzsche Part 1, The (De unge år: Erik Nietzsche)"
    />
  </div>
</section>
`;
